File: Frontend/src/domain/billing/invoice.ts
export const InvoiceStatus = {
  Draft: 1,
  Sent: 2,
  Paid: 3,
  Overdue: 4,
  Cancelled: 5,
  PartiallyPaid: 6,
} as const;

export type InvoiceStatus = (typeof InvoiceStatus)[keyof typeof InvoiceStatus];

export type InvoiceSummary = {
  id: string;
  invoiceNumber: string;
  status: InvoiceStatus;
  clientId: string;
  clientCompanyName: string;
  invoiceDate: string;
  dueDate: string;
  total: number;
  createdAt: string;
};

export type InvoiceLineItem = {
  id: string;
  itemId?: string | null;
  description: string;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
};

export type InvoiceDetail = {
  id: string;
  invoiceNumber: string;
  status: InvoiceStatus;
  clientId: string;
  clientCompanyName: string;
  clientContactName: string;
  clientEmail: string;
  invoiceDate: string;
  dueDate: string;
  subtotal: number;
  taxRate: number;
  taxAmount: number;
  total: number;
  notes?: string | null;
  createdAt: string;
  updatedAt?: string | null;
  lineItems: InvoiceLineItem[];
};

export { invoiceStatusClass, invoiceStatusLabel } from "./invoice-labels";

File: Backend/DataAccess/BillFlow.Models/Entities/Invoice.cs
using BillFlow.Models.Shared.Enums;

namespace BillFlow.Models.Entities;

public class Invoice
{
    public Guid Id { get; set; }

    public Guid OwnerId { get; set; }

    public User Owner { get; set; } = null!;

    public Guid ClientId { get; set; }

    public Client Client { get; set; } = null!;

    public string InvoiceNumber { get; set; } = null!;

    public InvoiceStatus Status { get; set; } = InvoiceStatus.Draft;

    public DateTime InvoiceDate { get; set; }

    public DateTime DueDate { get; set; }

    public decimal Subtotal { get; set; }

    public decimal TaxRate { get; set; }

    public decimal TaxAmount { get; set; }

    public decimal Total { get; set; }

    public string? Notes { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public ICollection<InvoiceLineItem> LineItems { get; set; } = new List<InvoiceLineItem>();

    public ICollection<Payment> Payments { get; set; } = new List<Payment>();

    public ICollection<InvoiceShareToken> ShareTokens { get; set; } = new List<InvoiceShareToken>();
}
